### DISCUSSION TITLE
bias-variance-decomp hates this dataset

### DISCUSSION THREAD

**Discussion by tomas-o-dev** [rasbt/mlxtend#821] [Q&A]
Odd but true - the red-wine quality dataset from https://archive.ics.uci.edu/ml/datasets/wine+quality
consistently raises an index error - summarised trace from running it with the sample code from the documentation
and opening it with df = pd.read_csv('../datasets/winequality-red.csv', sep=';')

D:\conda\lib\site-packages\mlxtend\evaluate\bias_variance_decomp.py in bias_variance_decomp
---> 86         X_boot, y_boot = _draw_bootstrap_sample(rng, X_train, y_train)
D:\conda\lib\site-packages\mlxtend\evaluate\bias_variance_decomp.py in _draw_bootstrap_sample
---> 16     return X[bootstrap_indices], y[bootstrap_indices]
D:\conda\lib\site-packages\pandas\core\series.py
D:\conda\lib\site-packages\pandas\core\indexing.py
KeyError: "Passing list-likes to .loc or [] with any missing labels is no longer supported.

it is a useful function that normally works  - this has me stumped!

**Comment by rasbt** [comment]
I am not sure that I wrote the function with pandas data frames in mind, so it might have been a coincidence that it worked before :). Not saying I don't want it to work with data frames as well, but it would require some tweaks, I assume. Until then, passing NumPy arrays should work:

I.e., via the `.values` attribute. E.g.,

```
avg_expected_loss, avg_bias, avg_var = bias_variance_decomp(
        tree, X_train.values, y_train.values, X_test.values, y_test.values, 
        loss='0-1_loss',
        random_seed=123)
```

**Reply by tomas-o-dev** [reply]
Thank you for your patience - I got all focused on a data problem and did not think about is as a data structure problem
Don't know how I missed this in the scikit docs: if you put a dataframe into test_train_split, it returns dataframes ,,,
Me, I wouldn't touch the code, just update the docs: 

one can use pd.DataFrame.to_numpy() to convert, or perhaps more usefully

from sklearn.preprocessing import MinMaxScaler  # all scikit scalers return ndarray
Xtrain = MinMaxScaler().fit_transform(X_train)
Xtest = MinMaxScaler().transform(X_test)

## bias_variance_decomp() REQUIRES NUMERIC TARGETS
from sklearn.preprocessing import LabelEncoder
ytrain = LabelEncoder().fit_transform(y_train)
ytest = LabelEncoder().fit_transform(y_test)

**Reply by rasbt** [reply]
Thanks for the feedback. I just modified the function (https://github.com/rasbt/mlxtend/pull/832) so that it will now raise a ValueError + explanation in the future when a DataFrame is passed -- at least until I had time to add DataFrame support.
